### PR TITLE
leading coefficient stuff

### PR DESCRIPTION
**Pull request by tthsqe12**
fixes https://github.com/oscar-system/Singular.jl/issues/403
fixes https://github.com/oscar-system/Singular.jl/issues/398
fixes https://github.com/oscar-system/Singular.jl/issues/350

### CODE REVIEW

File: src/poly/poly.jl [hunk 1]
```diff
@@ -193,20 +193,34 @@ function order(p::spoly)
...
+function leading_exponent_vector(p::spoly)
    R = parent(p)
    n = nvars(R)
    A = Array{Int}(undef, n)
    libSingular.p_GetExpVL(p.ptr, A, R.ptr)
    return A
 end
 
+function lead_exponent(p::spoly)
+   @warn "please use leading_exponent_vector instead of lead_exponent"
```
Comment (fingolfin) [inline src/poly/poly.jl:211]: Should this be a deprecation, using `@deprecate`?

**Review by fingolfin — APPROVED** [review]
Looks good to me. Does this need to wait for anything else, or can it just be merged?

File: src/poly/poly.jl (hunk 1 above)
Comment (tthsqe12) [reply src/poly/poly.jl:211]: I do not understand `@deprecate` and recently have had problems with it hanging. I would like to keep it this way and then eventually remove it when it is surely gone from all of the other packages.

Comment (thofma) [reply src/poly/poly.jl:211]: Let me give it a try.

**Comment by thofma** [comment]
I pushed the `@deprecate` use and will merge as soon as everything is green.

**Pull request merged**